plonky2/src/plonk/plonk_common.rs

```diff
@@ -138,6 +139,25 @@ where
     sum
 }
 
+pub fn reduce_with_powers_circuit<F: RichField + Extendable<D>, const D: usize>(
+    builder: &mut CircuitBuilder<F, D>,
+    terms: &[Target],
+    alpha: Target,
+) -> Target {
+    if terms.len() <= ArithmeticGate::new_from_config(&builder.config).num_ops + 1 {
+        terms
+            .iter()
+            .rev()
+            .fold(builder.zero(), |acc, &t| builder.mul_add(alpha, acc, t))
+    } else {
+        let terms_ext = terms
```

Comment:
nit: `reduce_with_powers_ext_circuit` will lead to a `ReducingFactorTarget::reduce` call, but seems like it could be preferable to call `ReducingFactorTarget::reduce_base` so that it can potentially use `ReducingGate` rather than `ReducingExtensionGate`